ordered_product/tests/test_ordered_product_page.py

```diff
@@ -0,0 +1,132 @@
+        assert response.status_code == 302
+        template_names = set(template.origin.template_name for template in response.templates)
+        assert template_names == set()
+
+    @pytest.mark.django_db()
+    def test_order(self, client):
+        response = client.get('/ordered_product/new_order/')
+        assert response.status_code == 200
+        template_names = set(template.origin.template_name for template in response.templates)
+        assert 'ordered_product/new_order.html' in template_names
+
+    @pytest.mark.django_db()
+    def test_valid_search_product(self, client):
+        response = client.get('/ordered_product/new_order/search_product', {'product_name': VALID_PRODUCT_NAME})
+        assert response.status_code == 200
+        assert response.context['message'] == ""
+        assert response.context['supplier_products'] != []
+
+    @pytest.mark.django_db()
+    def test_invalid_search_product(self, client):
+        response = client.get('/ordered_product/new_order/search_product', {'product_name': INVALID_PRODUCT_NAME})
+        assert response.status_code == 200
+        assert response.context['message'] == "No products found."
+        assert response.context['supplier_products'] == []
+
+    @pytest.mark.django_db()
+    def test_valid_find_delivery(self, client):
+        response = client.get('/ordered_product/new_order/find_delivery/'+VALID_SUPPLIER_PRODUCT_ID)
+        assert response.status_code == 200
+        template_names = set(template.origin.template_name for template in response.templates)
+        assert 'ordered_product/new_order.html' in template_names
+        assert response.context['sup_pro_id'] == int(VALID_SUPPLIER_PRODUCT_ID)
+        supplier_product = SupplierProduct.objects.get(supplier_product_id=int(VALID_SUPPLIER_PRODUCT_ID))
+        assert set(response.context['deliveries']) == set(DeliveryLocation.objects.filter(
+                                                          user_name=supplier_product.user_name))
+
+    @pytest.mark.django_db()
+    def test_invalid_find_delivery(self, client):
+        response = client.get('/ordered_product/new_order/find_delivery/'+INVALID_SUPPLIER_PRODUCT_ID)
+        assert response.status_code == 302
+
+    @pytest.mark.django_db()
+    def test_valid_order_summary(self, client):
+        url = '/ordered_product/new_order/order_summary/'+VALID_SUPPLIER_PRODUCT_ID+','+VALID_DELIVERY_LOCATION_ID
+        response = client.get(url)
+        assert response.status_code == 200
+        template_names = set(template.origin.template_name for template in response.templates)
+        assert 'ordered_product/new_order.html' in template_names
+        assert response.context['supplier_product'] == SupplierProduct.objects.get(
+                                                       supplier_product_id=int(VALID_SUPPLIER_PRODUCT_ID))
+        assert response.context['delivery_location'] == DeliveryLocation.objects.get(id=int(VALID_DELIVERY_LOCATION_ID))
```

Comment:
<relevant for all the tests> : do keep in mind, future-wise, that the tests depend on specific DB fields to existing with a particular, specifically mentioned, name like `supplier_product` or `delivery_location`. if/when there will be a need for the smallest change in one of those - there will be a LOT of maintenance to do.

Reply:
Now I see that most of those are dict keys you create when calling the render method. that is actually fine. I'll leave the comment here anyway though, so keep it in the back of your mind. :)